docs/tutorials/plugin-provider.md

```diff
@@ -0,0 +1,15 @@
+# Plugin provider
+
+The Plugin provider is a provider that allows DNS providers to integrate with ExternalDNS via an HTTP interface. The Plugin provider implements the Provider interface but instead of implementing code specific to a provider, it implements and HTTP client that sends request to an HTTP API. The idea behind this is that providers can be implemented in a separate program that exposes an HTTP API that the Plugin provider can interact with. The ideal setup for providers is to run as sidecars of the ExternalDNS container and listen on localhost only, but this is not strictly a requirement even if we do not recommend other setups.
```

Comment:
```suggestion
The provider "plugin" allows to integrate ExternalDNS with DNS providers via an HTTP interface.
The provider "plugin" implements the Provider interface. Instead of implementing code specific to a provider, it implements an HTTP client that sends request to an HTTP API. 
The idea behind this is, that providers can be implemented in a separate program. These programs expose an HTTP API that the Plugin provider can interact with. The ideal setup for providers is to run as sidecars in the same pod as the ExternalDNS container and listen on localhost only. This is not strictly a requirement, still we would not recommend other setups.
```

Tried to make this a bit more readable by shortening the sentences.

Reply:
Sounds legit, but I will refactor this instead of committing it as I want to have some changes.